### PR TITLE
Add deepface models to pipeline. 

### PR DESCRIPTION
**Pull request by judithvdw**

### CODE REVIEW

File: lr_face/models.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    def __init__(self, resolution=(100, 100)):
+        self.resolution = resolution
+
+    def fit(self, X, y):
+        assert X.shape[1:3] == self.resolution
+        pass
+
+    def predict_proba(self, X):
+        assert X.shape[2:4] == self.resolution
```
Comment (rolfypma) [inline lr_face/models.py]: wil je dit hard eisen? zou m uitcommenten

File: lr_face/utils.py [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+import math
+
+import cv2
+import numpy as np
+import pandas as pd
+from PIL import Image
+from keras.preprocessing import image
+
+
+def distance(a, b):
```
Comment (rolfypma) [inline lr_face/utils.py]: ik zou niet je eigen euclidische afstand interpreteren. neem bijv numpy.linalg.norm ?

File: lr_face/utils.py [hunk 3]
```diff
@@ -171,3 +188,15 @@ def concat_columns(df, column_names, output_column_name, separator='-'):
...
+
+
+def resize(img, target_size):
+        right_size_img  = cv2.resize(img, target_size)
+
+        img_pixels = image.img_to_array(right_size_img)
+        img_pixels = np.expand_dims(img_pixels, axis=0)
+
+        # normalize input in [0, 1]
+        img_pixels /= 255
```
Comment (rolfypma) [inline lr_face/utils.py]: waarom is normalizen nodig in een resize functie? klinkt als een aparte functie, of een resize_and_normalize functie?

Comment (judithvdw) [reply lr_face/utils.py]: Dit is nog een leftover van de enorme decect/crop/align/resize functie uit deelface, ik heb alleen het laatste stukje daarvan genomen. Zal hem hernoemen.

File: lr_face/utils.py (hunk 2 above)
Comment (judithvdw) [reply lr_face/utils.py]: De hele functie kan weg, want die werd überhaupt niet gebruikt.

File: lr_face/models.py (hunk 1 above)
Comment (judithvdw) [reply lr_face/models.py]: Dit is jouw code ;) ik wil hem best uitcommenten.

**Comment by rolfypma** [comment]
kan je nog de requirements updaten?

File: params.py [hunk 4]
```diff
@@ -74,9 +73,13 @@
 For the input of an experiment the 'current_set_up' list can be updated
 """
 SCORERS = {
-    'current_set_up': ['test'],
+    'current_set_up': ['openface', 'vggface', 'dummy'],
     'all': {
-        'test': TestModel()
+        'dummy': DummyModel(),
+        'openface': Deepface_Lib_Model(model=OpenFace.loadModel()),
```
Comment (rolfypma) [inline params.py]: deze implementatie zorgt er trouwens wel voor dat al deze modellen geladen worden, ook as ze niet gebruikt worden. Onhandig als laden duur is?

File: lr_face/models.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        scores = []
+        for pair in X:
+            img1 = resize_and_normalize(pair[0], self.model.input_shape[1:3])
+            img2 = resize_and_normalize(pair[1], self.model.input_shape[1:3])
+
+            img1_representation = self.model.predict(img1)[0, :]
+            img2_representation = self.model.predict(img2)[0, :]
+
+            score = spatial.distance.cosine(img1_representation, img2_representation)
+            scores.append([1-score, score])
```
Comment (rolfypma) [inline lr_face/models.py]: is deze omdraaien niet logischer? Dan is p[:,1] groot als h1 waar is
